### PR TITLE
feat(cx): introduce Side Nav

### PR DESCRIPTION
**Pull request by InnaAtanasova**
## Description
The current PR is a refactor for Side Nav which has a new design

!!! IMPORTANT !!! Note:
When reviewing please switch to Horizon theme or use this link: https://deploy-preview-3913--fundamental-styles.netlify.app/?path=/docs/components-side-navigation--cozy-wide-icons&globals=theme:sap_horizon

### CODE REVIEW

**Comment by InnaAtanasova** [comment]
When reviewing please use this link or switch to Horizon theme
https://deploy-preview-3913--fundamental-styles.netlify.app/?path=/docs/components-side-navigation--cozy-wide-icons&globals=theme:sap_horizon

File: src/styles/nested-list.scss [hunk 1]
```diff
@@ -5,378 +5,253 @@ $block: #{$fd-namespace}-nested-list;
...
   @include fd-reset();
 
-  list-style: none;
   width: 100%;
-  border-bottom: var(--sapList_BorderWidth) solid var(--sapList_BorderColor);
+  list-style: none;
 
-  & [class*="level"] {
-    border-bottom: none;
+  &.level-1 {
```
Comment (platon-rov) [inline src/styles/nested-list.scss:35]: is it correct according to the bem?

File: src/styles/nested-list.scss [hunk 2]
```diff
@@ -5,378 +5,253 @@ $block: #{$fd-namespace}-nested-list;
...
-        padding-left: 0;
-      }
     }
   }
 
-  &__group-header {
+  &__arrow {
     @include fd-reset();
+    @include fd-flex-center();
+    @include set-width(1rem);
```
Comment (platon-rov) [inline src/styles/nested-list.scss:202]: prefix?

**Review by platon-rov — APPROVED** [review]
Wow, nice!

**Comment by mikerodonnell89** [comment]
The popovers have popover arrows in the designs, and I think it'd be beneficial to show the whole side nav for the examples where the popover is shown rather than just showing the popover

**Comment by mikerodonnell89** [comment]
In the specs there are 8px from the top of the nav to the top of the first item, and 8px from the bottom of the bottom item to the bottom of the nav. In the side nav in this PR we have 8px on the top then 12px on the bottom, I think due to the whole sidenav having different paddings for top/bottom.

**Comment by mikerodonnell89** [comment]
Docs are missing the vertical overflow and filter examples but that can come later and I don't think that may be totally agnostic from this component

File: src/styles/nested-list.scss (hunk 1 above)
Comment (mikerodonnell89) [reply src/styles/nested-list.scss:35]: These class names are being added programmatically in ngx in `nested-list.directive.ts`. Maybe time for a rename but it'd have to be done on both ngx and styles

Comment (mikerodonnell89) [reply src/styles/nested-list.scss:35]: These class names are being added programmatically in ngx in `nested-list.directive.ts`. Maybe time for a rename, it'd have to be done on both ngx and styles

Comment (InnaAtanasova) [reply src/styles/nested-list.scss:35]: we have in fundamental-styles similar cases with the classes `is-open`, `has-child`, etc. 
`level-1`, `level-2`, etc are not modifier classes in the sense of BEM. They don't modify the properties of the element they are attached to, but rather to its child elements.
In BEM we have something like: `fd-nested-list  fd-nested-list--transparent`. This modifies the background of ``fd-nested-list`, but in our case `level-2` would modify the item and the title, which are child and grandchild of the element. In the past we wanted to distinguish such behaviour and that's why classes like `is-open`, `has-child`, `level-*` appeared. I am open to discussion about that. Not a problem at all to rename the classes.

**Comment by InnaAtanasova** [comment]
> More of a docs thing, and just a suggestion - The popovers have popover arrows in the designs, and I think it'd be beneficial to show the whole side nav for the examples where the popover is shown rather than just showing the popover

In fundamental-styles we don't have popover body with arrow placed like this. We can do this in fund-ngx
Screen Shot 2022-10-11 at 12 45 28 PM

**Comment by InnaAtanasova** [comment]
> Designs state that the first level always shows an icon (I guess that cannot be enforced on our end)
> 
> Screen Shot 2022-10-06 at 10 30 25 AM

Yes, I addressed this question to the designers as well, because it's written that Level 1 should always have icons but in the specs there's this one:
Screen Shot 2022-10-11 at 12 48 27 PM

**Comment by InnaAtanasova** [comment]
> Docs are missing the vertical overflow and filter examples but that can come later and I don't think that may be totally agnostic from this component

I am not sure if they are part of the final design. There's no design for the vertical overflow buttons and in the "Backstage Design Specs" the overflow is shown with a scrollbar:
Screen Shot 2022-10-11 at 1 02 50 PM

File: src/styles/nested-list.scss (hunk 2 above)
Comment (InnaAtanasova) [reply src/styles/nested-list.scss:202]: I opened a PR for renaming the mixins missing the prefix. Depending on which PR is merged first I will rebase and update. 
https://github.com/SAP/fundamental-styles/pull/3921

File: src/styles/nested-list.scss (hunk 1 above)
Comment (mikerodonnell89) [reply src/styles/nested-list.scss:35]: Thanks for the explanation, I think just leave them as-is then. If we do want to change, let's make a separate issue and discuss

**Review by mikerodonnell89 — CHANGES_REQUESTED** [review]
Apologies for nit-picking but in the specs looks like there should be 12px of padding on top of the cozy divider and 16px of the bottom
Screen Shot 2022-10-12 at 12 11 50 PM

**Review by mikerodonnell89 — APPROVED** [review]

**Review by mikerodonnell89 — CHANGES_REQUESTED** [review]
If this gets merged, and we update the fundamental-styles version in fd-ngx, anyone using the quartz/fiori theme will have a broken side nav, even if we make the required API updates. I think we may need a new component entirely with a new name

**Review by mikerodonnell89 — APPROVED** [review]

**Comment by droshev** [comment]
We need to keep the old implementation in styles and introduce the new one in CX package

**Review by droshev — CHANGES_REQUESTED** [review]
who will move this to the `cx` packages. BTW the previous version of the side nav should stay in `fundamental-styles` deprecated and pointing to the new implementation

**Review by droshev — APPROVED** [review]

**Pull request merged**